java/ql/src/experimental/semmle/code/java/Logging.qll

```diff
@@ -28,6 +27,18 @@ class LoggingCall extends MethodAccess {
       m.getReturnType() instanceof VoidType and
       this = m.getAReference()
     )
+    or
+    exists(RefType t, Method m | t.hasQualifiedName("android.util", "Log") |
+      (
+        m.hasName(["d", "e", "i", "v", "w", "wtf"])
+      ) and
+      (
+        m.getDeclaringType().getASourceSupertype*() = t or
+        m.getDeclaringType().extendsOrImplements*(t)
```

Comment:
These seem like they're saying the same thing, are you aware of a case that means you need both?

Reply:
To be honest I just tried to keep this as close to the originally approved query as possible. Why are these required in lines 24 and 25? Doesn't getDeclaringType() already handle inheritance and interface implementation?

Reply:
There are some differences in relation to generics, in particular how these relations deal with parameterizations, raw types, type variables, and wildcard types. But for something simple like this case, then I expect them to be equivalent (with some possible caveats if there are generic subclasses of `android.util.Log`).

Reply:
I simplified the query considering the fact the Log class is final. All my test results remained the same.